src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameService.java

```diff
@@ -243,10 +250,72 @@ public Game getGameFromRandomUser(User user) {
     PageRequest pageRequest = PageRequest.of(pageIndex, 1);
 
     // get the page with the game
-    Page<Game> somePage = gameRepository.getOtherUserWithActiveGameThatWasNotPlayedOn(pageRequest, user.getId(), user,
-        user.getGender(), minAgeDate, maxAgeDate, user.getBlockedUsers(), user.getMatchedUsers());
+    Page<Game> somePage = gameRepository.getOtherUserWithActiveGameThatWasNotPlayedOn(
+      pageRequest, 
+      user.getId(), 
+      user,
+      user.getGender(), 
+      minAgeDate, 
+      maxAgeDate, 
+      user.getBlockedUsers(),
+      user.getMatchedUsers()
+    );
+
+    // subset the page of users to only users that have a haversine distance of less than user.getMaxRange()
+    List<Game> games = somePage.getContent();
+    List<Game> gamesWithDistance = new ArrayList<>();
+    // go over all games that match the other criteria
+    for(Game game : games){
```

Comment:
page always contains ONE game. So either you increase the size of the pageRequest, or you find another solution^^

Reply:
One possibility would be to get all the games with the constraints (there shouldn't be many anyways, especially with the small user base (if the app becomes a success, we must refactor things anyway)), and then use your solution (but maybe add some tolerance, so that it's not too restricting^^).

Reply:
> page always contains ONE game. So either you increase the size of the pageRequest, or you find another solution^^

thanks, I did not know that!